Mathlib/Probability/Distributions/Gamma.lean

```diff
@@ -0,0 +1,195 @@
+    ∫ (t : ℝ) in Ioi 0, (r ^ a / Gamma a) * t ^ (a - 1) * exp (-(r * t))
+    = 1 := by
+  calc ∫ (t : ℝ) in Ioi 0, (r ^ a / Gamma a) * t ^ (a - 1) * exp (-(r * t))
+  _ = ∫ (t : ℝ) in Ioi 0, (r ^ a / Gamma a) • (t ^ (a - 1) * exp (-(r * t))) := by
+    congr
+    ext x
+    rw [smul_eq_mul, mul_assoc]
+  _ = (r ^ a / Gamma a) • ∫ (t : ℝ) in Ioi 0, t ^ (a - 1) * exp (-(r * t)) := by
+    apply integral_smul
+  _ = (r ^ a / Gamma a) * (1 / r) ^ a * Gamma (a) := by
+    rw [integral_rpow_mul_exp_neg_mul_Ioi ha hr, smul_eq_mul, mul_assoc]
+  _ = 1 := by
+    rw [mul_assoc, div_mul_eq_mul_div, ← mul_assoc, mul_div_assoc,
+      div_self (Gamma_pos_of_pos ha).ne', mul_one, div_rpow zero_le_one hr.le, one_rpow,
+      mul_one_div, div_self (rpow_pos_of_pos hr _).ne']
+
+open Measure
+
+/-- The Pdf of the gamma Distribution integrates to 1-/
+@[simp]
+lemma lintegral_gammaPdf_eq_one (a r : ℝ) (ha : 0 < a) (hr : 0 < r) :
+    ∫⁻ x, gammaPdf a r x = 1 := by
+  rw [lintegral_eq_lintegral_Ici_add_Iio (gammaPdf a r) 0, ← ENNReal.toReal_eq_one_iff]
+  have leftSide : ∫⁻ x in Iio 0, gammaPdf a r x = 0 := by
+    rw [set_lintegral_congr_fun measurableSet_Iio
+      (ae_of_all _ (fun x (hx : x < 0) ↦ gammaPdf_of_neg hx)), lintegral_zero]
+  have rightSide :
+      ∫⁻ x in Ici 0, gammaPdf a r x =
+      ∫⁻ x in Ici 0, ENNReal.ofReal (r ^ a / (Gamma a) * x ^ (a-1) * exp (-(r * x))) :=
+    set_lintegral_congr_fun isClosed_Ici.measurableSet
+      (ae_of_all _ (fun x (hx : 0 ≤ x) ↦ gammaPdf_of_nonneg hx))
+  simp only [leftSide, add_zero]
+  rw [rightSide]
+  rw [← integral_eq_lintegral_of_nonneg_ae]
+  · rw [MeasureTheory.integral_Ici_eq_integral_Ioi, pow_exp_integral_Ioi_insert_1 ha hr]
+  · unfold EventuallyLE
+    simp only [Pi.zero_apply]
+    rw [ae_restrict_iff]
+    simp only [mem_Ici]
+    · apply ae_of_all
+      intro x hx
+      positivity
+    · refine measurableSet_le measurable_const ?_
+      refine Measurable.mul ?_ ?_
+      · measurability
+      · apply (Measurable.neg ?_).exp
+        exact measurable_id.const_mul _
+  · rw [← restrict_Ioi_eq_restrict_Ici]
+    refine ContinuousOn.aestronglyMeasurable ?_ measurableSet_Ioi
+    refine ContinuousOn.mul ?_ (Continuous.continuousOn (by continuity))
+    exact continuousOn_const.mul (continuousOn_id.rpow_const fun x hx ↦ Or.inl (ne_of_gt hx))
+
+end GammaPdf
+
+open MeasureTheory
+
+/-- Measure defined by the Gamma Distribution -/
+noncomputable
+def gammaMeasure (a r : ℝ) : Measure ℝ :=
+  volume.withDensity (gammaPdf a r)
+
+lemma IsProbabilityMeasureGamma {a r : ℝ} (ha : 0 < a) (hr : 0 < r) :
+    IsProbabilityMeasure (gammaMeasure a r) where
+  measure_univ := by simp [gammaMeasure, lintegral_gammaPdf_eq_one a r ha hr]
+
+section GammaCdf
+
+/-- CDF of the Gamma Distribution -/
+noncomputable
+def gammaCdfReal (a r : ℝ) : StieltjesFunction :=
+  cdf (gammaMeasure a r)
+
+lemma gammaCdfReal_eq_integral (a r x : ℝ) (ha : 0 < a) (hr : 0 < r) :
```

Comment:
`a` and `r` can be implicitized (and `x` should probably be the last argument)

Reply:
Will do!